lib/Onnxifi/Base.h

```diff
@@ -112,9 +154,13 @@ class Graph {
   /// Run inference.
   onnxStatus run();
 
+  /// Run async using thread pool.
+  static void runAsync(std::function<void(void)> fn);
+
 private:
   BackendPtr backendPtr_;
   Function *function_;
+  static ThreadPool threadPool_;
```

Comment:
I think the thread pool itself needs to live on the backend level. This way we can limit the number of threads for some of the accelerators and make bigger for CPU and Interpreter backend.